dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,257 @@
+                    }
+                }
+                return arraysize;
+            } else {
+                int objectsize = sizeof(cls);
+                for (Field f : getAllNonStaticFields(obj.getClass())) {
+                    Class<?> fcls = f.getType();
+                    if (fcls.isPrimitive())
+                        continue;
+                    f.setAccessible(true);
+                    ref = f.get(obj);
+                    if (ref != null && !isCalculated(calculated, ref)) {
+                        calculated.put(ref, ref);
+                        int referentSize = retainedSize(ref, calculated);
+                        objectsize += referentSize;
+                    }
+                }
+                return objectsize;
+            }
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SuppressWarnings("restriction")
+    public static int sizeof(Class<?> cls) {
+
+        if (cls == null)
+            throw new NullPointerException();
+
+        if (cls.isArray())
+            throw new IllegalArgumentException();
+
+        if (cls.isPrimitive())
+            return primsize(cls);
+
+        int lastOffset = Integer.MIN_VALUE;
+        Class<?> lastClass = null;
+
+        for (Field f : getAllNonStaticFields(cls)) {
+            if (Modifier.isStatic(f.getModifiers()))
+                continue;
+
+            int offset = (int) UnsafeAccess.UNSAFE.objectFieldOffset(f);
+            if (offset > lastOffset) {
+                lastOffset = offset;
+                lastClass = f.getClass();
+            }
+        }
+        if (lastOffset > 0)
+            return modulo8(lastOffset + primsize(lastClass));
+        else
+            return 16;
+    }
+
+    private static Field[] getAllNonStaticFields(Class<?> cls) {
+        if (cls == null)
+            throw new NullPointerException();
+
+        List<Field> fieldList = new ArrayList<Field>();
+        while (cls != Object.class) {
+            for (Field f : cls.getDeclaredFields()) {
+                if (!Modifier.isStatic(f.getModifiers()))
+                    fieldList.add(f);
+            }
+            cls = cls.getSuperclass();
+        }
+        Field[] fs = new Field[fieldList.size()];
+        fieldList.toArray(fs);
+        return fs;
+    }
+
+    private static boolean isCalculated(HashMap<Object, Object> calculated, Object test) {
+        Object that = calculated.get(test);
+        return that != null && that == test;
+    }
+
+    private static int primsize(Class<?> cls) {
+        if (cls == byte.class)
+            return 1;
+        if (cls == boolean.class)
+            return 1;
+        if (cls == char.class)
+            return 2;
+        if (cls == short.class)
+            return 2;
+        if (cls == int.class)
+            return 4;
+        if (cls == float.class)
+            return 4;
+        if (cls == long.class)
+            return 8;
```

Comment:
Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)